### DISCUSSION TITLE
How to set caffeine to support sized based and weight based eviction at same time?

### DISCUSSION THREAD

**Discussion by lfygh** [ben-manes/caffeine#1340] [Q&A] [answered]
I want size based and weight based eviction works at the same time?   is it possible use native caffeine?

**Comment by ben-manes** [comment, marked as answer]
Nope, you may limit capacity by either the entry count or total weight of all entries. Internally they are the same by using a weight of 1 per entry for a size limit. When would you need both?

**Reply by lfygh** [reply]
We use mmap to read files. In order to save the overhead of mmaping the same file multiple times, we cache the mmap buffer of the file. Therefore, we want to limit the memory size of mmap and also limit the number of mmap files.

**Reply by ben-manes** [reply]
Interesting. You could try to manually prune the size with the help of the Cache.policy().eviction() methods to find the entries to discard. Thus weight would be automatically enforced and size manually. Would that be acceptable?

**Reply by lfygh** [reply]
Of course, we are currently using this method. Thank you for your reply.
